src/DotNetEnv/Extensions/LinqExtensions.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace DotNetEnv.Extensions
+{
+    public static class LinqExtensions
+    {
+        public static Dictionary<string, string> ToDictionary(this IEnumerable<KeyValuePair<string, string>> @this, CreateDictionaryOption option = CreateDictionaryOption.Default)

```

Comment:
Keep the existing behavior of `TakeLast`, please. It was chosen extremely deliberately to match the behavior you would expect if you set an envelope var multiple times in the shell. I.e. if you wrote
```
EVDNE=a
EVDNE=b
```
You would absolutely expect `echo $EVDNE` to output `b` and not `a` and definitely, definitely not throw an error.

Reply:
The current behavior allows you to merge multiple settings together (theoretically, and separate from this library's functionality), such that you might have dupes, and expect that the last one written in the file, same as if you were writing them in the shell, is the one you would expect to apply.

Reply:
I do understand your wish to leave the current behaviour as it is, but that is exactly why I do not want to use this library. It add a extension to a standard type, which introduces unexpected behavior for a default case (not DotEnv).

If you just use ToDictionary you would expect it to throw exceptions on duplicates.
Since we cannot restrict usage to the DotEnv-UseCases, we should be aware of the expected behaviour on this type in general.

If someone has it's own ToDictionary() defined (because it is really annoying to type this whole select on KeyValuePair...) he may runs into bugs, due to unexpectedly using the wrong extension (which is a really subtle difference, just a wrong using).

A workaround could be to change the ExtensionMethods name for example. That is a breaking change too, but it will give some compiler-warning due to method does not exist anymore.
For example: `ToDotEnvDictionary()` (with or without options, your choice; with options I'm okay with default TakeLast then).